### DISCUSSION TITLE
CRUD: currency columun

### DISCUSSION THREAD

**Discussion by blondie63** [Laravel-Backpack/community-forum#873] [Q&A (Help)] [answered]
How can i set a column to show a currency fields with correct format for Italy ?
<img width="1205" alt="BKP6 x-currency" src="https://github.com/Laravel-Backpack/community-forum/assets/1261851/aa739817-174e-4693-adf9-880a7ff05fc6">

i'm using BackPack v6

Thanks for help :)

**Comment by karandatwani92** [comment]
Hey @blondie63 

Please try the available options of the [number](https://backpackforlaravel.com/docs/6.x/crud-columns#number-1) column.
OR You'll need to create a custom column if it doesn't fit your needs.
* Command to create new column type `php artisan backpack:column currency`
* Guide on [Creating a Custom Column Type](https://backpackforlaravel.com/docs/6.x/crud-columns#creating-a-custom-column-type-1)

**Reply by blondie63** [reply]
I've set my column so:
```
        $this->crud->addColumn('importo', [
            'label' => 'Importo',
            'type' => 'number',
            'prefix'        => '€ ',
            'suffix'        => ' EUR',
            'decimals'      => 2,
            'dec_point'     => ',',
            'thousands_sep' => '.',
        ]);
```
but the result is this:
<img width="148" alt="BKP6x-format-column-issue" src="https://github.com/Laravel-Backpack/community-forum/assets/1261851/6360eb43-b9ad-42f8-8680-9b7851733065">

I cleared the cache but nothing, it doesn't respect the formatting. What can I check?

**Reply by karandatwani92** [reply]
Hey @blondie63 

I find your code working for me:
```
CRUD::column([
            'name' => 'total',
            'label' => 'Importo',
            'type' => 'number',
            'prefix'        => '€ ',
            'suffix'        => ' EUR',
            'decimals'      => 2,
            'dec_point'     => ',',
            'thousands_sep' => '.',
        ]);
```

![image](https://github.com/Laravel-Backpack/community-forum/assets/8214221/6f25cc36-1bfb-43c6-9fff-8ec8ca0ad691)

The version I'm using:
`backpack/crud: 6.5.2`

Please re-check; I suspect there is something on your end.

**Comment by pxpm** [comment]
I think your column definition is wrong @blondie63 

```php
 $this->crud->addColumn([
            'name' => 'importo',
            'label' => 'Importo',
            'type' => 'number',
            'prefix'        => '€ ',
            'suffix'        => ' EUR',
            'decimals'      => 2,
            'dec_point'     => ',',
            'thousands_sep' => '.',
        ]);
```
Should work.

**Reply by blondie63** [reply]
ok this working for me ! Thanks
@pxpm Do you know how can i set a right aligment for this column ? Normally numbers must be right aligned :)

**Reply by blondie63** [reply]
Hi @pxpm what about right alignment ?

**Reply by pxpm** [reply]
Sorry @blondie63 , you should be able to align the column using css classes on the column wrapper.

**Reply by blondie63** [reply]
ok, but it could be a useful parameter to add as standard, what do you think?

**Reply by pxpm** [reply, marked as answer]
Sorry for the delay on response @blondie63 

You mean something like: `'textAlign => 'right'` in the column definition ? I think that's something that could be useful yes, but only if we can do it via column wrapper (we only need to change one file). If that requires changes in ALL columns, it may be more difficult to justify the effort for such small change. 

Cheers

**Reply by blondie63** [reply]
ok anyway i've used column wrapper 
Cheers
